File: Python/Level3/네트워크.py
def network(now, n, computers, visited):
    visited[now] = 1
    for i in range(n):
        if computers[now][i] == 1 and visited[i] == 0:
            visited[i] = 1
            network(i, n, computers, visited)


def solution(n, computers):
    answer = 0
    visited = [0] * n
    for i in range(n):
        if visited[i] == 0:
            network(i, n, computers, visited)
            answer += 1
    return answer

File: programmers/C++/Level3/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

void network(int now, vector<vector<int>> computers, vector<bool>& visited) {
	
	visited[now] = true;

	for (int i = 0; i < computers.size(); i++)
		if (i != now && computers[now][i] == 1 && visited[i] == false)
			network(i, computers, visited);
}

int solution(int n, vector<vector<int>> computers) {

	int answer = 0;
	vector<bool> visited(n, false);

	for (int i = 0; i < n; i++)
	{
		if (visited[i] == false)
		{
			network(i, computers, visited);
			answer++;
		}
	}

	return answer;
}
